### ISSUE TITLE
Build failure under Windows using GCC-9 after inclusion of C++11'isms

### ISSUE THREAD

**Issue by krcroft** [dreamer/dosbox-staging#42]
The Windows GCC-9.x build started failing (https://github.com/dreamer/dosbox-staging/runs/299604881) on commit: https://github.com/dreamer/dosbox-staging/commit/4e65c8f69fd808f252e5a25b5df5e3ad643ebd34.

This build turns on `-std=c++11` despite this is commit not touching `configure.ac` or the Makefiles.  

It's as if `configure` inspected the headers/used-features and decided to enable the `-std=c++11` flag (not that this should cause any problems! It certainly is fine under the other environments).

Perhaps there's some broken code in the WIN32 path(s) that need some cleanup to make compatible with GCC C++11.

**Comment by dreamer** [comment]
OK, I meandered throughout the code before finding the solution. What I suspect happened here: WIN32 is user-defined macro (as it does not start with `_`); before C++11 it was probably predefined by compilers or something. Before C++11 was finalized, Clang probably purged this define earlier than GCC (that's why the build started failing only on Clang once you turned on C++0x), and GCC postponed it for the final version of the standard. As for MSVC, this define is injected in a project file (either automatically at creation or when I upgraded solution files to the newest Visual Studio - I am quite sure I haven't put it in there myself).
